jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -57,6 +57,7 @@
 public class Content
 {
     private static final Logger LOG = LoggerFactory.getLogger(Content.class);
+    private static final ByteBuffer EMPTY_READ_ONLY_BUFFER = BufferUtil.EMPTY_BUFFER.asReadOnlyBuffer();
```

Comment:
Do we really need this?
The buffer is empty, and any attempt to write would result in a `BufferOverflowException` anyway.

Reply:
It's needed by `Content.Chunk.asReadOnly()` to avoid needlessly duplicating error/eof/empty chunks, as is tested by `ContentTest.testAsReadOnly()`.

Reply:
Can you expand the test to avoid read-onlyness if remaining() == 0?

Reply:
Ok, done.